### PR TITLE
Add support for OPVault pwd history, custom attributes and (fix) attachments

### LINKED ISSUE #5669: Incomplete import of 1Password OPVault
## Overview
KeepassXC doesn't appear to fully import an `opvault` file that was created via 1Password version 4. Specifically, the following fields are not imported:
- Attributes
- Attachments
- Password History

## Steps to Reproduce
1. Load the attached opvault, which has an attribute of `test`, an attachment called `test` and a historical password of `test`.
2. Observe the error log.

## Expected Behavior
- All of the above fields are loaded.

## Actual Behavior
- The file is partially imported

## Context
It appears that attachments aren't being extracted properly because the OPVault file fails the HMAC check. Commenting out the return [here](https://github.com/keepassxreboot/keepassxc/blob/develop/src/format/OpVaultReaderAttachments.cpp#L107) appears to allow the rest of the code to run correctly and to import the attachments.

Not sure about the other two field types, but it seems like they're simply not supported? If there is interest, I can investigate that and submit a PR.

[test.opvault.zip](https://github.com/keepassxreboot/keepassxc/files/5498891/test.opvault.zip)

Debug info:
```
KeePassXC - Version 2.7.0-snapshot
Build Type: Snapshot
Revision: 91dea9c

Qt 5.12.8
Debugging mode is disabled.

Operating system: Ubuntu 20.04.1 LTS
CPU architecture: x86_64
Kernel: linux 5.4.0-52-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
 libgcrypt 1.8.5
```
Operating System: Linux
Desktop Env: XFCE
Windowing System: X11

### PR DESCRIPTION
**Pull request by kiwiz**
Fixes #5669 

## Testing strategy
Only tested with a simple vault. Probably still needs tests, but opening this up for feedback.

## Type of change
- ✅ New feature (change that adds functionality)

### CODE REVIEW

File: src/format/OpVaultReaderAttachments.cpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ bool OpVaultReader::readAttachment(const QString& filePath,
     OpData01 icon01;
     if (!icon01.decode(iconBytes, itemKey, itemHmacKey)) {
         qCritical() << "Unable to decipher attachment icon in " << filePath << ": " << icon01.errorString();
-        return false;
+        // return false;
```
Comment (kiwiz) [inline src/format/OpVaultReaderAttachments.cpp:108]: Bad idea, of course.

**Comment by kiwiz** [comment]
Added some tests (that are currently failing). I noticed that `fillFromSection` causes a new password history entry to be added to the list, which seems unexpected. Additionally, it seems that `fillFromSectionField` already handles importing attributes, as long as the `v` attribute exists. It seems like 1Password 4 might not support attributes, hence the inconsistency?

Not sure what the best approach should be here.

**Comment by droidmonkey** [comment]
Thank you for this work but it will not be merged. We only support 1Password version 7 or higher.

**Comment by kiwiz** [comment]
That's unfortunate, but understandable. Is it worth adding a note about this to the docs (if one doesn't already exist)? In either case, thanks for looking!

**Pull request closed without merging**